sqlmesh/core/state_sync/base.py

```diff
@@ -291,3 +366,33 @@ def unpause_snapshots(
             unpaused_dt: The datetime object which indicates when target snapshots
                 were unpaused.
         """
+
+    def migrate(self) -> None:
+        """Migrate the state sync to the latest SQLMesh / SQLGlot version."""
+        row = self.get_versions(validate=False)
+
+        if row:
+            migration_version, sqlglot_version = row
+            remote_sqlglot_version = major_minor(sqlglot_version)
+            migrations = MIGRATIONS[migration_version:]
+        else:
+            remote_sqlglot_version = major_minor(SQLGLOT_VERSION)
+            migrations = MIGRATIONS
+            self.init_schema()
```

Comment:
Shouldn't the content of `init_schema` be the first script in the migrations package?

Reply:
no it can't be because that could mess up further schemas. init_schema is not static